azureDatabaseUtils/src/main/java/bio/terra/workspace/azureDatabaseUtils/database/DatabaseService.java

```diff
@@ -160,20 +198,29 @@ public void pgRestore(
     List<String> commandList = generateCommandList("psql", dbName, dbHost, dbPort, adminUser);
     Map<String, String> envVars = Map.of("PGPASSWORD", determinePassword());
 
-    localProcessLauncher.launchProcess(commandList, envVars);
-
     logger.info(
-        "Running DatabaseService.pgRestore on file {} from blob container: {}",
-        blobFileName,
-        blobContainerName);
+            "Streaming DatabaseService.pgRestore input from blob file {} in container: {}",
+            blobFileName,
+            blobContainerName);
 
-    storage.streamInputFromBlobStorage(
-        localProcessLauncher.getOutputStream(),
-        blobFileName,
-        blobContainerName,
-        blobContainerUrlAuthenticated);
+    localProcessLauncher.launchProcess(commandList, envVars);
+
+    // Stream wrapping is confusing, so in English: blob storage > base 64 decoding > decryption > local process input
+    // NB we use the pipes to receive blob storage via an output stream and convert that into an input stream for the subsequent processing
+    // NB we use a separate thread to handle the download so that we can be downloading and sending to the thread in parallel
```

Comment:
I might have questions about this later, but I think I get it sufficiently for review purposes